### ISSUE TITLE
potential typo in `postgres` user service account email detection

### ISSUE THREAD

**Issue by kriswuollett** [terraform-google-modules/terraform-google-sql-db#456]
### TL;DR

The is service account check in the `postgres` module doesn't appear to be technically correct because something like `oscar@notgserviceaccount.com` would be detected as a service account, and which email is possible to be created because it is not the `gserviceaccount.com` domain:

https://github.com/terraform-google-modules/terraform-google-sql-db/blob/15298c275e2d531e3379d1c445b2317c99e6363b/modules/postgresql/main.tf#L32

Perhaps it was a typo because further down in the code a leading `.` is used:

https://github.com/terraform-google-modules/terraform-google-sql-db/blob/15298c275e2d531e3379d1c445b2317c99e6363b/modules/postgresql/main.tf#L270-L274

Not the place where it may be necessary, but technically also checking for proper [Project ID](https://cloud.google.com/resource-manager/docs/creating-managing-projects) could be done with a regex which I think the following would be sufficient:

`@[a-z][a-z0-9-]{4,28}[a-z0-9]\.iam\.gserviceaccount\.com$`.

I assume applying the terraform with an email account like `oscar@notgserviceaccount.com` would fail anyways because I assume GCP wouldn't find that service account.

### Terraform Configuration

```hcl
N/A
```

### Terraform Version

```sh
N/A
```
